sentence_transformers/sparse_encoder/models/SpladePooling.py

```diff
@@ -45,32 +55,77 @@ def __init__(
         Returns:
             Dictionary containing SPLADE pooled embeddings
         """
-        # Get the MLM head logits (shape: batch_size, seq_length, vocab_size)
         mlm_logits = features["token_embeddings"]
+        attention_mask = features["attention_mask"]  # Shape: [batch_size, seq_length]
 
-        # Apply ReLU and log transformation for SPLADE
-        if self.activation_function == "relu":
-            splade_scores = torch.log1p(torch.relu(mlm_logits))
-        elif self.activation_function == "log1p_relu":
-            splade_scores = torch.log1p(torch.log1p(torch.relu(mlm_logits)))
-        else:
-            raise ValueError("activation_function must be either 'relu' or 'log1p_relu'")
+        # Unsqueeze attention_mask to be [batch_size, seq_length, 1] for broadcasting
+        attention_mask_expanded = attention_mask.unsqueeze(-1).to(mlm_logits.dtype)
+
+        batch_size, seq_len, vocab_s = mlm_logits.shape
+        device = mlm_logits.device
 
-        # Pool across sequence length dimension
+        # Initialize pooled scores based on pooling strategy
         if self.pooling_strategy == "max":
-            pooled_scores = torch.max(splade_scores, dim=1)[0]  # shape: batch_size, vocab_size
-        else:  # sum
-            pooled_scores = torch.sum(splade_scores, dim=1)  # shape: batch_size, vocab_size
+            pooled_scores = torch.full((batch_size, vocab_s), float("-inf"), dtype=mlm_logits.dtype, device=device)
+        elif self.pooling_strategy == "sum":
+            pooled_scores = torch.zeros((batch_size, vocab_s), dtype=mlm_logits.dtype, device=device)
+        else:
+            raise ValueError(f"Unsupported pooling_strategy: {self.pooling_strategy}")
+
+        # Process in chunks if chunk_size is set, otherwise process the entire sequence at once
+        chunk_size = seq_len if self.chunk_size is None else self.chunk_size
+
+        for i in range(0, seq_len, chunk_size):
+            try:
+                current_chunk_logits = mlm_logits[:, i : i + chunk_size, :]
+                current_chunk_mask = attention_mask_expanded[:, i : i + chunk_size, :]
+
+                masked_current_chunk_logits = current_chunk_logits * current_chunk_mask
+
+                if self.activation_function == "relu":
+                    if not self.training:
+                        current_chunk_transformed = masked_current_chunk_logits.relu_().log1p_()
+                    else:
+                        current_chunk_transformed = torch.log1p(torch.relu(masked_current_chunk_logits))
+                elif self.activation_function == "log1p_relu":
+                    if not self.training:
+                        current_chunk_transformed = masked_current_chunk_logits.relu_().log1p_().log1p_()
+                    else:
+                        current_chunk_transformed = torch.log1p(torch.log1p(torch.relu(masked_current_chunk_logits)))
```

Comment:
```suggestion
                if not self.training:
                    current_chunk_transformed = masked_current_chunk_logits.relu_().log1p_()
                else:
                    current_chunk_transformed = masked_current_chunk_logits.relu().log1p()
                # With "log1p_relu", we apply a second log1p
                if self.activation_function == "log1p_relu":
                    if not self.training:
                        current_chunk_transformed = current_chunk_transformed.log1p_()
                    else:
                        current_chunk_transformed = current_chunk_transformed.log1p()
```
What do you think of this simplification?

Reply:
It is indeed simplifying the code, so i like it !